lib/api/1.1/northbound/authentication.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express'),
+    passport = require('passport'),
+    JwtStrategy = require('passport-jwt').Strategy,
+    LocalStrategy = require('passport-local').Strategy;
+
+module.exports = authRouterFactory;
+
+di.annotate(authRouterFactory, new di.Provide('Http.Api.Auth'));
+di.annotate(authRouterFactory, new di.Inject(
+        'Auth.Services'
+        )
+);
+
+function authRouterFactory (
```

Comment:
I'd suggest replacing this with an `authMiddlewareFactory' that returns an express middleware function.  Then inject this function in the router factories that need it.  See https://github.com/RackHD/on-http/compare/master...brianparry:feature/swagger-auth#diff-87f6ea9f0a13097e18945b4d56c26dd5R1.